<html>
<h2>Protein progress</h2>

<h3>EC number classification</h3>
<h4>Description of the task</h4>
</br>
<h4>Image for multiple datasets<h4>
<img src="images/EC_prediction/stats_EC40_EC50_DEEPre_ECPred.png"/>

<h4>Paper to process and extract result</h4>
<ul>
 <li><a href="https://academic.oup.com/bioinformatics/article/36/8/2401/5698270" >UDSMProt: universal deep sequence models for protein classification </a></li>
 <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6349967/" >mlDEEPre: Multi-Functional Enzyme Function Prediction With Hierarchical Multi-Label Deep Learning</a></li>
 <li><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2368-y" > ECPred: a tool for the prediction of the enzymatic functions of protein sequences based on the EC nomenclature</a></li>
 <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6030869/" > DEEPre: sequence-based enzyme EC number prediction by deep learning</a></li>
 <li><a href="https://academic.oup.com/bioinformatics/article-abstract/36/17/4583/5843785" > </a>HECNet: a hierarchical approach to enzyme function classification using a Siamese Triplet Network </li>
 <li><a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=9086514" > </a>The Classification of Enzymes by Deep Learning</li>
</ul>


<h4>On EC40 dataset</h4>
Description of the dataset :</br>
The corresponding dataset with similarity threshold40% relying on clusters from a prior CD-HIT [Fu et al., 2012] run is termedEC40. Compared toDEEPre, we slightly adapted their dataset generationprocedure by clustering Swiss-Prot all at once with CD-HIT, instead ofclustering enzymes and non-enzymes separately, as this mimics the wayUniRef50 clusters are constructed, albeit for a similarity threshold 40%in  this  case.  The  dataset  using  a  similarity  threshold  of  50%  by  usingUniRef50-clusters is termed EC50. In both cases, we do not balance thenumber of enzymes and non-enzymes and restrict ourselves to proteinswith a single EC-label.

The  EC50  and  EC40  EC  classification  datasets  are  constructedaccording to the following procedure:</br>
1.   Acquire Swiss-Prot (2016_08 for EC40 and 2017_03 for EC50).</br>
2.   Cluster Swiss-Prot with CD-HIT (40% similarity cutoff) for EC40 oracquire UniRef50 (2017_03) for EC50 and apply it to Swiss-Prot.</br>
3.   Remove non-enzymes which have not enough experimental evidencein order to avoid misleading false negatives (annotated evidence isgreater or equal to 4).</br>
4.   Remove proteins which are annotated as fragments.</br>
5.   Remove  enzymes  with  multiple  enzymatic  annotations  in  order  toobtain a single-label classification problem.</br>
6.   Filter proteins by length to include only proteins with more than 50and less than 5000 amino acids.</br>
7.   Split by clusters into 80% training and 10% validation and 10% testset.</br>
8.   For test and validation set, we select only representatives (from CD-HIT or UniRef50 clustering) or alternatively the first member of thecluster in the case where the representative was excluded by filteringrules. Optionally a similar reduction is applied to the training set toobtain a set of non-redundant sequences.</br>
9.   Use  the  predefined  cluster  assignments  of  the  previous  step  andsimilarly distribute all remaining Swiss-Prot clusters onto training,validation and test dataset according to split ratios 90%, 5% and 5%,respectively, to obtain a clean train-test-split on the whole Swiss-Prot dataset consistent with chosen clusters for the downstream classifica-tion task. Again, we keep only representatives for validation and testset.The last step is important for suitable database creation for PSI-BLASTfor the experiments in Section 3.2.2 and Appendix 6. One thing can alreadybe anticipated, as a result, for significantly more sequences from the testdataset PSI-BLAST returned empty queries resulting in less informativefeatures for the test set.</br>
</br>

Result on test set :</br>
 <table class="result">
  <tr>
    <th>Model</th>
    <th>Accuracy on lvl0</th>
    <th>Accuracy on lvl1</th>
    <th>Accuracy on lvl2</th>
    <th>Paper/Source</th>
    <th>Code</th>
  </tr>
  <tr>
    <td>UDSMProt</td>
    <td>0.91</td>
    <td>0.87</td>
    <td>0.84</td>
    <td><a href="https://academic.oup.com/bioinformatics/article/36/8/2401/5698270" >UDSMProt: universal deep sequence models for protein classification </a></td>
    <td><a href="https://github.com/nstrodt/UDSMProt">Official</a></td>
  </tr>
  <tr>
    <td>My ensemble network</td>
    <td>--</td>
    <td>--</td>
    <td>0.95</td>
    <td>Not available</td>
    <td>Not available</td>
  </tr>
</table>


<h4>On DEEPre dataset</h4>
Description of the dataset :</br>
- similarity threshold 40%

</br>
Result on test set : </br>
 <table class="result">
  <tr>
    <th>Model</th>
    <th>Accuracy on lvl0</th>
    <th>Accuracy on lvl1</th>
    <th>Accuracy on lvl2</th>
    <th>Paper/Source</th>
    <th>Code</th>
  </tr>
  <tr>
    <td>UDSMProt</td>
    <td>0.87</td>
    <td>0.84</td>
    <td>0.78</td>
    <td><a href="https://academic.oup.com/bioinformatics/article/36/8/2401/5698270" >UDSMProt: universal deep sequence models for protein classification </a></td>
    <td><a href="https://github.com/nstrodt/UDSMProt">Official</a></td>
  </tr>
  <tr>
    <td>DEEPre (seqÃ¾PSSM)</td>
    <td>0.88</td>
    <td>0.82</td>
    <td>0.43</td>
    <td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6349967/" >mlDEEPre: Multi-Functional Enzyme Function Prediction With Hierarchical Multi-Label Deep Learning</a></td>
    <td></td>

  </tr>
</table>


<h4>On ECPred dataset</h4>
Description of the dataset :</br>
- similarity threshold 50%

</br>
Result on test set : </br>
 <table class="result">
  <tr>
    <th>Model</th>
    <th>Mean F1 on lvl0</th>
    <th>Mean F1 on lvl1</th>
    <th>Paper/Source</th>
    <th>Code</th>
  </tr>
  <tr>
    <td>UDSMProt</td>
    <td>0.98</td>
    <td>0.95</td>
    <td><a href="https://academic.oup.com/bioinformatics/article/36/8/2401/5698270" >UDSMProt: universal deep sequence models for protein classification </a></td>
    <td><a href="https://github.com/nstrodt/UDSMProt">Official</a></td>
  </tr>
  <tr>
    <td>ECPred</td>
    <td>0.96</td>
    <td>0.96</td>
    <td><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2368-y" ></a>ECPred: a tool for the prediction of the enzymatic functions of protein sequences based on the EC nomenclature</td>
    <td></td>

  </tr>
</table>


<style>
.result, th, td {
  border: 1px solid black;
}

</style>

</html>
